nearcore/src/lib.rs

```diff
@@ -167,6 +167,11 @@ fn apply_store_migrations_if_exists(
         info!(target: "near", "Migrate DB from version 30 to 31");
         migrate_30_to_31(store_opener, &near_config)?;
     }
+    if db_version <= 31 {
+        // version 31 => 32: DBCol::TransactionResult now uses a merge operator.
+        // Migration is no-op, but we need to bump the version, because
+        // db_version 31 binary can't open db_version 32 db.
```

Comment:
You need to set version here, i.e.:

```rust
    let store = store_opener.open()?.get_store(Temperature::Hot);
    near_store::version::set_store_version(&store, 32)?;
```

The reason it’s not set when migrating from 27 to 28 is because when
we do that migration we jump to migration from 28 to 29 which sets the
version straight to 29.